examples/go.mod

```
@@ -1,6 +1,6 @@
 module go.mongodb.org/atlas-sdk/v20231115008/examples
 
-go 1.22.1
+go 1.21.1
```

should we use the split version?
```suggestion
go 1.21
toolchain 1.21.0
```
can't really find the difference and if it matters as a lib

---

presence of toolchain is local but once present go mod tidy makes the bump. 
Toolchain is not required for SDK

---

Removed unneded patch version. Presence of toolchain triggers automatic updates so I'm reluctant to add it for now.